### ISSUE TITLE
Can't comunicate with the module SIM800

### ISSUE THREAD

**Issue by fdeizaguirre** [Gabriellavoura/Mqtt-and-Sim800-900#4]
The arduino can't talk with the SIM800 module.

I'm usign the mqtt_basic_pub_and_sub example and I include HardwareSerial.h under an Arduino Mega.

I put an Serial.println in attempt to watch the answer of the module when the library sim800 sends an clean "AT", and the answer in spite of be "OK" is "⸮"
I change several times the ports on the arduino with the same behaviour.

What can I'm doing wrong?
I'll really apresiate your help.

**Comment by Gabriellavoura** [comment]
Let's start with basics:
1. You should verify if the **baudrate** in the prompt is set **equal** to the **Serial.Begin** declaration in the code.
2. Remember that ports **0** and **1** of Arduino MEGA are used **exclusively** by serial communication with USB you can't connect anything to them.  

Try it and let me know if it has worked.

**Comment by fdeizaguirre** [comment]
Gabriel, 
Thanks for your replay.
1. If you are talking about the serial com between PC-Arduino yes I'm using the same baudrate. Or should I also set the baud rate for Serial1 as Serial1.begin?
2. I'm using TX1(18)/RX1(19)
I hope I don't missunderstud you.

**Comment by Gabriellavoura** [comment]
1. Yes, you should try use the same baud rate for hardware and software serial.

**Comment by quoctl** [comment]
help !!, can you guide the connection between arduino and SIM800 
please!!

**Comment by imranaalam** [comment]
Have you tried to run the Sim800l through usb to uart;once you are sure that every thing works fine then  use  this library.

**Comment by rhidwan** [comment]
@fdeizaguirre
I had same problem and I figured out that:
-- I was powering the gsm module only from the arduino. It didn't work at this time. When I connect it with external power everything works fine.

I don't know if it will solve your problem. But may be It will help someone.

**Comment by fdeizaguirre** [comment]
Dears.
 @rhidwan thank you very much. All this time I migrated to another solution and just now I came back to this. 
My trouble was similar to yours. I was using external power, but it couldn't supply the 2A that the SIM800 needs at the connection. So I solved first using a 3000uF capacitor in parallel to the SIM800. And after that, I used the adaptation circuit that is suggested at SIM800 Hardware Design doc. Page 20 Power Supply https://simcom.ee/documents/SIM800/SIM800_Hardware Design_V1.09.pdf
Once again, thank you all. I hope this thread helps someone else.
